Dockerfile

```diff
@@ -14,6 +14,8 @@ COPY main.go main.go
 COPY api/ api/
 COPY controllers/ controllers/
 COPY pkg/   pkg/
```

Comment:
The only thing that is currently resides in `pkg` directory is `client` package. We can't directly move `client` pkg inside `core` because it uses `api/v1beta1` to intialize scheme. I have refactor it into `internal` and `core` package.
I keep client pkg to refactor till last because it could cause some issue which are hard to debug. May be I could raise seperate PR for it after this one merged.

Reply:
I had created below jira to incorporate this review comment.
https://issues.redhat.com/browse/KOGITO-4408